packages/liveblocks-react-ui/src/primitives/AiMessage/index.tsx

```diff
@@ -72,20 +81,19 @@ function ToolInvocation({
     [ai, chatId, messageId, part.status, part.toolName, part.toolCallId]
   );
 
-  if (tool === undefined || tool.render === undefined) return null;
-  const RenderFn = tool.render as ComponentType<
-    AiToolInvocationProps<JsonObject, ToolResultData>
-  >;
+  const props = useMemo(() => {
+    const { type: _, ...rest } = part;
+    return {
+      ...rest,
+      respond,
+      types: undefined as never,
+      [kInternal]: {
+        execute: tool?.execute,
+      },
+    };
+  }, [part, respond, tool?.execute]);
```

Comment:
This is not strictly part of the fix, but I noticed that we were passing it in as `value={props}` to the context provider below, which I believe would cause any component subscribing to that context would get re-rendered, even if none of the fields actually changed.

Reply:
Happy to revert this or discuss it elsewhere if you think it's controversial or not doing anything, @marcbouchenoire.

Reply:
I have nothing against this 👍